// @whop/sdk/resources/ai-chats.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.AIChats = void 0;
const resource_1 = require("../core/resource.js");
const pagination_1 = require("../core/pagination.js");
const path_1 = require("../internal/utils/path.js");
class AIChats extends resource_1.APIResource {
    /**
     * Creates a new AI chat
     *
     * Required permissions:
     *
     * - `ai_chat:create`
     */
    create(body, options) {
        return this._client.post('/ai_chats', { body, ...options });
    }
    /**
     * Fetches a single AI chat by ID
     */
    retrieve(id, options) {
        return this._client.get((0, path_1.path) `/ai_chats/${id}`, options);
    }
    /**
     * Updates an AI chat
     *
     * Required permissions:
     *
     * - `ai_chat:update`
     */
    update(id, body = {}, options) {
        return this._client.patch((0, path_1.path) `/ai_chats/${id}`, { body, ...options });
    }
    /**
     * Fetches all AI chats for the current user
     */
    list(query = {}, options) {
        return this._client.getAPIList('/ai_chats', (pagination_1.CursorPage), { query, ...options });
    }
    /**
     * Deletes an AI chat
     *
     * Required permissions:
     *
     * - `ai_chat:delete`
     */
    delete(id, options) {
        return this._client.delete((0, path_1.path) `/ai_chats/${id}`, options);
    }
}
exports.AIChats = AIChats;
