// @whop/sdk/resources/index.js
"use strict";
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
Object.defineProperty(exports, "__esModule", { value: true });
exports.Withdrawals = exports.Webhooks = exports.Verifications = exports.Users = exports.Transfers = exports.Topups = exports.SupportChannels = exports.Shipments = exports.SetupIntents = exports.Reviews = exports.Refunds = exports.Reactions = exports.PromoCodes = exports.Products = exports.Plans = exports.PayoutMethods = exports.Payments = exports.PaymentMethods = exports.Notifications = exports.Messages = exports.Memberships = exports.Members = exports.LedgerAccounts = exports.Leads = exports.Invoices = exports.Forums = exports.ForumPosts = exports.Files = exports.FeeMarkups = exports.Experiences = exports.Entries = exports.DmMembers = exports.DmChannels = exports.Disputes = exports.Courses = exports.CourseStudents = exports.CourseLessons = exports.CourseLessonInteractions = exports.CourseChapters = exports.CompanyTokenTransactions = exports.Companies = exports.CheckoutConfigurations = exports.ChatChannels = exports.AuthorizedUsers = exports.Apps = exports.AppBuilds = exports.AccountLinks = exports.AccessTokens = exports.AIChats = void 0;
const tslib_1 = require("../internal/tslib.js");
tslib_1.__exportStar(require("./shared.js"), exports);
var ai_chats_1 = require("./ai-chats.js");
Object.defineProperty(exports, "AIChats", { enumerable: true, get: function () { return ai_chats_1.AIChats; } });
var access_tokens_1 = require("./access-tokens.js");
Object.defineProperty(exports, "AccessTokens", { enumerable: true, get: function () { return access_tokens_1.AccessTokens; } });
var account_links_1 = require("./account-links.js");
Object.defineProperty(exports, "AccountLinks", { enumerable: true, get: function () { return account_links_1.AccountLinks; } });
var app_builds_1 = require("./app-builds.js");
Object.defineProperty(exports, "AppBuilds", { enumerable: true, get: function () { return app_builds_1.AppBuilds; } });
var apps_1 = require("./apps.js");
Object.defineProperty(exports, "Apps", { enumerable: true, get: function () { return apps_1.Apps; } });
var authorized_users_1 = require("./authorized-users.js");
Object.defineProperty(exports, "AuthorizedUsers", { enumerable: true, get: function () { return authorized_users_1.AuthorizedUsers; } });
var chat_channels_1 = require("./chat-channels.js");
Object.defineProperty(exports, "ChatChannels", { enumerable: true, get: function () { return chat_channels_1.ChatChannels; } });
var checkout_configurations_1 = require("./checkout-configurations.js");
Object.defineProperty(exports, "CheckoutConfigurations", { enumerable: true, get: function () { return checkout_configurations_1.CheckoutConfigurations; } });
var companies_1 = require("./companies.js");
Object.defineProperty(exports, "Companies", { enumerable: true, get: function () { return companies_1.Companies; } });
var company_token_transactions_1 = require("./company-token-transactions.js");
Object.defineProperty(exports, "CompanyTokenTransactions", { enumerable: true, get: function () { return company_token_transactions_1.CompanyTokenTransactions; } });
var course_chapters_1 = require("./course-chapters.js");
Object.defineProperty(exports, "CourseChapters", { enumerable: true, get: function () { return course_chapters_1.CourseChapters; } });
var course_lesson_interactions_1 = require("./course-lesson-interactions.js");
Object.defineProperty(exports, "CourseLessonInteractions", { enumerable: true, get: function () { return course_lesson_interactions_1.CourseLessonInteractions; } });
var course_lessons_1 = require("./course-lessons.js");
Object.defineProperty(exports, "CourseLessons", { enumerable: true, get: function () { return course_lessons_1.CourseLessons; } });
var course_students_1 = require("./course-students.js");
Object.defineProperty(exports, "CourseStudents", { enumerable: true, get: function () { return course_students_1.CourseStudents; } });
var courses_1 = require("./courses.js");
Object.defineProperty(exports, "Courses", { enumerable: true, get: function () { return courses_1.Courses; } });
var disputes_1 = require("./disputes.js");
Object.defineProperty(exports, "Disputes", { enumerable: true, get: function () { return disputes_1.Disputes; } });
var dm_channels_1 = require("./dm-channels.js");
Object.defineProperty(exports, "DmChannels", { enumerable: true, get: function () { return dm_channels_1.DmChannels; } });
var dm_members_1 = require("./dm-members.js");
Object.defineProperty(exports, "DmMembers", { enumerable: true, get: function () { return dm_members_1.DmMembers; } });
var entries_1 = require("./entries.js");
Object.defineProperty(exports, "Entries", { enumerable: true, get: function () { return entries_1.Entries; } });
var experiences_1 = require("./experiences.js");
Object.defineProperty(exports, "Experiences", { enumerable: true, get: function () { return experiences_1.Experiences; } });
var fee_markups_1 = require("./fee-markups.js");
Object.defineProperty(exports, "FeeMarkups", { enumerable: true, get: function () { return fee_markups_1.FeeMarkups; } });
var files_1 = require("./files.js");
Object.defineProperty(exports, "Files", { enumerable: true, get: function () { return files_1.Files; } });
var forum_posts_1 = require("./forum-posts.js");
Object.defineProperty(exports, "ForumPosts", { enumerable: true, get: function () { return forum_posts_1.ForumPosts; } });
var forums_1 = require("./forums.js");
Object.defineProperty(exports, "Forums", { enumerable: true, get: function () { return forums_1.Forums; } });
var invoices_1 = require("./invoices.js");
Object.defineProperty(exports, "Invoices", { enumerable: true, get: function () { return invoices_1.Invoices; } });
var leads_1 = require("./leads.js");
Object.defineProperty(exports, "Leads", { enumerable: true, get: function () { return leads_1.Leads; } });
var ledger_accounts_1 = require("./ledger-accounts.js");
Object.defineProperty(exports, "LedgerAccounts", { enumerable: true, get: function () { return ledger_accounts_1.LedgerAccounts; } });
var members_1 = require("./members.js");
Object.defineProperty(exports, "Members", { enumerable: true, get: function () { return members_1.Members; } });
var memberships_1 = require("./memberships.js");
Object.defineProperty(exports, "Memberships", { enumerable: true, get: function () { return memberships_1.Memberships; } });
var messages_1 = require("./messages.js");
Object.defineProperty(exports, "Messages", { enumerable: true, get: function () { return messages_1.Messages; } });
var notifications_1 = require("./notifications.js");
Object.defineProperty(exports, "Notifications", { enumerable: true, get: function () { return notifications_1.Notifications; } });
var payment_methods_1 = require("./payment-methods.js");
Object.defineProperty(exports, "PaymentMethods", { enumerable: true, get: function () { return payment_methods_1.PaymentMethods; } });
var payments_1 = require("./payments.js");
Object.defineProperty(exports, "Payments", { enumerable: true, get: function () { return payments_1.Payments; } });
var payout_methods_1 = require("./payout-methods.js");
Object.defineProperty(exports, "PayoutMethods", { enumerable: true, get: function () { return payout_methods_1.PayoutMethods; } });
var plans_1 = require("./plans.js");
Object.defineProperty(exports, "Plans", { enumerable: true, get: function () { return plans_1.Plans; } });
var products_1 = require("./products.js");
Object.defineProperty(exports, "Products", { enumerable: true, get: function () { return products_1.Products; } });
var promo_codes_1 = require("./promo-codes.js");
Object.defineProperty(exports, "PromoCodes", { enumerable: true, get: function () { return promo_codes_1.PromoCodes; } });
var reactions_1 = require("./reactions.js");
Object.defineProperty(exports, "Reactions", { enumerable: true, get: function () { return reactions_1.Reactions; } });
var refunds_1 = require("./refunds.js");
Object.defineProperty(exports, "Refunds", { enumerable: true, get: function () { return refunds_1.Refunds; } });
var reviews_1 = require("./reviews.js");
Object.defineProperty(exports, "Reviews", { enumerable: true, get: function () { return reviews_1.Reviews; } });
var setup_intents_1 = require("./setup-intents.js");
Object.defineProperty(exports, "SetupIntents", { enumerable: true, get: function () { return setup_intents_1.SetupIntents; } });
var shipments_1 = require("./shipments.js");
Object.defineProperty(exports, "Shipments", { enumerable: true, get: function () { return shipments_1.Shipments; } });
var support_channels_1 = require("./support-channels.js");
Object.defineProperty(exports, "SupportChannels", { enumerable: true, get: function () { return support_channels_1.SupportChannels; } });
var topups_1 = require("./topups.js");
Object.defineProperty(exports, "Topups", { enumerable: true, get: function () { return topups_1.Topups; } });
var transfers_1 = require("./transfers.js");
Object.defineProperty(exports, "Transfers", { enumerable: true, get: function () { return transfers_1.Transfers; } });
var users_1 = require("./users.js");
Object.defineProperty(exports, "Users", { enumerable: true, get: function () { return users_1.Users; } });
var verifications_1 = require("./verifications.js");
Object.defineProperty(exports, "Verifications", { enumerable: true, get: function () { return verifications_1.Verifications; } });
var webhooks_1 = require("./webhooks.js");
Object.defineProperty(exports, "Webhooks", { enumerable: true, get: function () { return webhooks_1.Webhooks; } });
var withdrawals_1 = require("./withdrawals.js");
Object.defineProperty(exports, "Withdrawals", { enumerable: true, get: function () { return withdrawals_1.Withdrawals; } });
